### DISCUSSION TITLE
Add a way to get a collection's name

### DISCUSSION THREAD

**Discussion by santiagopuentep** [meteor/meteor#11771] [Compat Packages]
Add a way of accessing a collections name, to use inside publications when publishing to it.

Currently there's a `Collection._name` field, but TypeScript complains and it's a hack.

So instead of doing `this.added("myCollection", _id, fields);` inside a publication, one could do `this.added(MyCollection.name, _id, fields);`.

**Comment by paulincai** [comment]
Hi @santiagopuentep,
could you please help me with a code sample that includes the relevant imports and the relevant code for your matter.

**Reply by santiagopuentep** [reply]
Simply create a collection:
`const myCollection = new Mongo.Collection("myCollection");`
And then you can access the name of the collection using `myCollection._name` but it's not an official way of doing so, so for example you don't get TypeScript types for it.

Useful inside a publication:

```
Meteor.publish("myPublication", function () {
	this.added(myCollection._name, "someId", {});

	this.ready();
});

```

instead of having to hardcode the name of the collection like this:

```
Meteor.publish("myPublication", function () {
	// here I have to use a string
	this.added("myCollection", "someId", {});

	this.ready();
});
```

We need an official way to access the name of the collection.

**Comment by StorytellerCZ** [comment]
Not exact, but related: https://github.com/dburles/mongo-collection-instances/issues/32
